### ISSUE TITLE
train yolov2, dataset VOC, the avg_obj is abnormal

### ISSUE THREAD

**Issue by louyanyang** [eric612/MobileNet-YOLO#94]
i train mobilenet_yolo_train.prototxt, but the "avg_obj" is low and the loss also is bad  all the time,  
this is log: 
I0404 16:52:24.429404 25214 solver.cpp:228] Iteration 36600, loss = 37.9715
I0404 16:52:24.429461 25214 solver.cpp:244]     Train net output #0: det_loss1 = 21.6224 (* 1 = 21.6224 loss)
I0404 16:52:24.429469 25214 solver.cpp:244]     Train net output #1: det_loss2 = 21.263 (* 1 = 21.263 loss)
I0404 16:52:26.339438 25214 sgd_solver.cpp:106] Iteration 36600, lr = 0.001
I0404 16:52:32.363132 25214 region_loss_layer.cpp:474] avg_noobj: 0.00336523 avg_obj: 0.0115073 avg_iou: 0.659157 avg_cat: 0.217839 recall: 0.875 class_count: 8
I0404 16:52:43.053544 25372 region_loss_layer.cpp:474] avg_noobj: 0.001208 avg_obj: 0.00842717 avg_iou: 0.441495 avg_cat: 0.312264 recall: 0.333333 class_count: 9
I0404 16:52:52.684368 25214 region_loss_layer.cpp:474] avg_noobj: 0.00489463 avg_obj: 0.00882935 avg_iou: 0.501176 avg_cat: 0.192338 recall: 0.4 class_count: 20
I0404 16:53:02.973937 25214 region_loss_layer.cpp:474] avg_noobj: 0.0013741 avg_obj: 0.003841 avg_iou: 0.323919 avg_cat: 0.399147 recall: 0.194444 class_count: 36
I0404 16:53:06.607053 25214 solver.cpp:228] Iteration 36610, loss = 49.6577
I0404 16:53:06.607105 25214 solver.cpp:244]     Train net output #0: det_loss1 = 23.4779 (* 1 = 23.4779 loss)
I0404 16:53:06.607115 25214 solver.cpp:244]     Train net output #1: det_loss2 = 29.1093 (* 1 = 29.1093 loss)
I0404 16:53:08.523295 25214 sgd_solver.cpp:106] Iteration 36610, lr = 0.001
I0404 16:53:14.958472 25372 region_loss_layer.cpp:474] avg_noobj: 0.00768608 avg_obj: 0.0218214 avg_iou: 0.532336 avg_cat: 0.326192 recall: 0.727273 class_count: 11
I0404 16:53:26.575536 25214 region_loss_layer.cpp:474] avg_noobj: 0.000791245 avg_obj: 0.00519375 avg_iou: 0.360174 avg_cat: 0.0821306 recall: 0.285714 class_count: 7
I0404 16:53:37.220430 25372 region_loss_layer.cpp:474] avg_noobj: 0.00567865 avg_obj: 0.014628 avg_iou: 0.632284 avg_cat: 0.260289 recall: 0.8 class_count: 10
I0404 16:53:50.293854 25372 region_loss_layer.cpp:474] avg_noobj: 0.000892151 avg_obj: 0.00312572 avg_iou: 0.333341 avg_cat: 0.308817 recall: 0.222222 class_count: 9

@eric612  can you help me analysing  the problem?

**Comment by NJUSTghw** [comment]
I meet the same issue when I train mobilenet_yolov3_train.prototxt. From the beginning of the training, the value of avg_obj began to drop rapidly to a small value, although the det_loss is declining. I use the mobilenet_yolov3_train.prototxt provided in the project. Can you help me with the problem?
I0331 21:02:57.352484  6984 sgd_solver.cpp:121] Iteration 20000, lr = 0.0005
I0331 21:02:59.023978  6984 yolov3_layer.cpp:362] noobj: 0.00465108 obj: 0.0301093 iou: 0.682898 cat: 0.219442 recall: 0.991071 recall75: 0.264286 count: 4
I0331 21:02:59.037346  6984 yolov3_layer.cpp:362] noobj: 0.000758756 obj: 0.00329237 iou: 0.654843 cat: 0.261659 recall: 0.885417 recall75: 0.1625 count: 4
I0331 21:02:59.051192  6984 yolov3_layer.cpp:362] noobj: 0.000244126 obj: 0.00195985 iou: 0.449462 cat: 0.297224 recall: 0.387879 recall75: 0.0181818 count: 1
I0331 21:03:02.365550  6984 yolov3_layer.cpp:362] noobj: 0.00317522 obj: 0.0203893 iou: 0.677549 cat: 0.196456 recall: 0.937037 recall75: 0.281481 count: 4
I0331 21:03:02.370246  6984 yolov3_layer.cpp:362] noobj: 0.000539571 obj: 0.00452065 iou: 0.606496 cat: 0.295081 recall: 0.905 recall75: 0.173571 count: 5
I0331 21:03:02.374889  6984 yolov3_layer.cpp:362] noobj: 0.000135505 obj: 0.000573044 iou: 0.546395 cat: 0.149488 recall: 0.533333 recall75: 0 count: 1
I0331 21:03:03.899535  6984 solver.cpp:252] Iteration 20010 (1.52742 iter/s, 6.54698s/10 iters), loss = 7.29454
I0331 21:03:03.899566  6984 solver.cpp:271]     Train net output #0: det_loss1 = 6.49158 (* 1 = 6.49158 loss)
I0331 21:03:03.899571  6984 solver.cpp:271]     Train net output #1: det_loss2 = 4.91641 (* 1 = 4.91641 loss)
I0331 21:03:03.899576  6984 solver.cpp:271]     Train net output #2: det_loss3 = 0.000598176 (* 1 = 0.000598176 loss)
I0331 21:03:03.899581  6984 sgd_solver.cpp:121] Iteration 20010, lr = 0.0005
I0331 21:03:04.285420  6984 yolov3_layer.cpp:362] noobj: 0.0052518 obj: 0.0389342 iou: 0.651559 cat: 0.181799 recall: 0.898809 recall75: 0.265079 count: 4
I0331 21:03:04.289974  6984 yolov3_layer.cpp:362] noobj: 0.00119811 obj: 0.0116849 iou: 0.626686 cat: 0.286476 recall: 0.886682 recall75: 0.227656 count: 3
I0331 21:03:04.294018  6984 yolov3_layer.cpp:362] noobj: 0.000299882 obj: 0.00294017 iou: 0.530596 cat: 0.467784 recall: 0.7 recall75: 0 count: 1
I0331 21:03:07.689951  6984 yolov3_layer.cpp:362] noobj: 0.00350664 obj: 0.0244414 iou: 0.667146 cat: 0.234929 recall: 0.908163 recall75: 0.221088 count: 4
I0331 21:03:07.701253  6984 yolov3_layer.cpp:362] noobj: 0.000717519 obj: 0.00477341 iou: 0.657951 cat: 0.255541 recall: 0.972222 recall75: 0.174868 count: 4
I0331 21:03:07.712463  6984 yolov3_layer.cpp:362] noobj: 0.00015041 obj: 0.00237985 iou: 0.47522 cat: 0.310172 recall: 0.475 recall75: 0 count: 3
I0331 21:03:11.754290  6984 yolov3_layer.cpp:362] noobj: 0.00261417 obj: 0.0165737 iou: 0.655013 cat: 0.174636 recall: 0.897778 recall75: 0.238889 count: 4
I0331 21:03:11.767717  6984 yolov3_layer.cpp:362] noobj: 0.000576303 obj: 0.0212369 iou: 0.656699 cat: 0.388044 recall: 0.921549 recall75: 0.272686 count: 8
I0331 21:03:11.780875  6984 yolov3_layer.cpp:362] noobj: 6.82836e-05 obj: 0.000834396 iou: 0.508117 cat: 0.601005 recall: 0.75 recall75: 0 count: 2
I0331 21:03:12.611395  6984 solver.cpp:252] Iteration 20020 (1.14788 iter/s, 8.71173s/10 iters), loss = 7.4897
I0331 21:03:12.611418  6984 solver.cpp:271]     Train net output #0: det_loss1 = 3.91782 (* 1 = 3.91782 loss)
I0331 21:03:12.611421  6984 solver.cpp:271]     Train net output #1: det_loss2 = 5.55278 (* 1 = 5.55278 loss)
I0331 21:03:12.611425  6984 solver.cpp:271]     Train net output #2: det_loss3 = 0.000135638 (* 1 = 0.000135638 loss)
I0331 21:03:12.611428  6984 sgd_solver.cpp:121] Iteration 20020, lr = 0.0005
I0331 21:03:14.301214  6984 yolov3_layer.cpp:362] noobj: 0.00383905 obj: 0.0258256 iou: 0.70261 cat: 0.22783 recall: 0.991667 recall75: 0.293492 count: 4
I0331 21:03:14.305927  6984 yolov3_layer.cpp:362] noobj: 0.00109361 obj: 0.0179543 iou: 0.622899 cat: 0.328089 recall: 0.770146 recall75: 0.103938 count: 3
I0331 21:03:14.310665  6984 yolov3_layer.cpp:362] noobj: 0.000156122 obj: 0.00177055 iou: 0.342293 cat: 0.496929 recall: 0.0666667 recall75: 0 count: 2
I0331 21:03:16.350148  6984 yolov3_layer.cpp:362] noobj: 0.00467738 obj: 0.0518846 iou: 0.675751 cat: 0.246547 recall: 0.9 recall75: 0.294444 count: 3
I0331 21:03:16.366036  6984 yolov3_layer.cpp:362] noobj: 0.000963213 obj: 0.00845974 iou: 0.603416 cat: 0.272836 recall: 0.856838 recall75: 0.0833333 count: 2
I0331 21:03:16.381867  6984 yolov3_layer.cpp:362] noobj: 0.000250786 obj: 0.00099018 iou: 0.447527 cat: 0.318402 recall: 0.333333 recall75: 0.05 count: 8

**Comment by eric612** [comment]
In sometimes , the training will not be convergence , especially when use concat layer instead of eltwise sum layer , I can't give you a good suggestion because the information were not enough , you may need upload the model prototxt , log files and what dataset you used.

**Comment by NJUSTghw** [comment]
the model prototxt I used is mobilenet_yolov3_train.prototxt [mobilenet_yolov3_train.txt](https://github.com/eric612/MobileNet-YOLO/files/3051191/mobilenet_yolov3_train.txt)
the dataset I used is PASCAL VOC 2007. Is the size of the input image not square which causes this error?

**Comment by eric612** [comment]
@NJUSTghw  , As I see , the reason is no pre-weights to be loaded .

**Comment by NJUSTghw** [comment]
thanks for your help. What should I do without a pre-trained model? How to get the pre-training model? or how can I train my pre-training model? @eric612

**Comment by lqian** [comment]
I meet the same issue. keep a watching!

**Comment by zouliangyu** [comment]
I have the same issue,too. And my loss come out 'negative' after several iterations.

**Comment by eric612** [comment]
Can you try this solver to test ? 
```
train_net: "models/yolov3/mobilenet_yolov3_lite_train.prototxt"
test_net: "models/yolov3/mobilenet_yolov3_lite_test.prototxt"
test_iter: 4952
test_interval: 1000
base_lr: 0.005
display: 10
max_iter: 120000
lr_policy: "multifixed"
gamma: 0.1
momentum: 0.9
weight_decay: 0.0005
snapshot: 1000
snapshot_prefix: "models/yolov3/mobilenet_yolov3_lite_deploy"
solver_mode: GPU
debug_info: false
snapshot_after_train: true
test_initialization: false
average_loss: 10
stepvalue: 80000
stepvalue: 100000

stagelr: 0.000001
stagelr: 0.00001
stagelr: 0.0001
stagelr: 0.001
stagelr: 0.0001
stagelr: 0.00001
stageiter: 50
stageiter: 100
stageiter: 200
stageiter: 80000
stageiter: 100000
stageiter: 120000
iter_size: 2
type: "SGD"
eval_type: "detection"
ap_version: "11point"
show_per_class_result: true
```

**Comment by qmzpg** [comment]
Delete all the batchNorm layers in the "mobilenetyolov3_train.prototxt", the avg_obj will rise soon, but it is still small finnally, at about 0.2~0.3 in my trainning process.
